### PR TITLE
InfoTooltip: refactor component to be accessible

### LINKED ISSUE #42384: Apply react-aria to InfoTooltip component
Apply react aria to `InfoTooltip (useTooltipTrigger)`

### PR DESCRIPTION
**Pull request by thisisobate**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Fixes #42384

**What's left**:
Update all usage of icon element within a tooltip to use icon buttons instead.

### CODE REVIEW

**Comment by thisisobate** [comment]
**New Findings**

Tooltips are designed to be focusable using the keyboard. In the case of the `infoTooltip`, it won't be an exception either. But there's a bit of a problem.

The way our tooltip component is designed within Grafana is that it requires that the child element must be a focusable element. InfoTooltip currently violates that.

The `infoTooltip` contains an `Icon` element, which makes it unaccessible using the keyboard. To make this focusable, we need to wrap our `Icon` element in a focusable element. Thankfully, we have the Button element to our rescue. 
Fixed it yeah? not really.

We need to somehow remove the custom button styling so as to achieve the same exact look as we had before.
And how do we do that? `ghostMode` property to the rescue!

`ghostMode` prop allows you to switch on/off the button style. Primary use case would be using icon-buttons within a tooltip.

**Comment by thisisobate** [comment]
oh wow...just figured we already have something similar within the codebase- IconButtons! 😍
https://github.com/grafana/grafana/blob/main/packages/grafana-ui/src/components/IconButton/IconButton.tsx

File: packages/grafana-ui/src/components/IconButton/IconButton.tsx [hunk 1]
```diff
@@ -49,9 +49,15 @@ export const IconButton = React.forwardRef<HTMLButtonElement, Props>(
   ) => {
     const theme = useTheme2();
     const styles = getStyles(theme, size, variant);
+    const tooltipString = React.isValidElement(tooltip) === true ? '' : (tooltip as string);
```
Comment (ashharrison90) [inline packages/grafana-ui/src/components/IconButton/IconButton.tsx:52]:
```suggestion
    const tooltipString = typeof tooltip === 'string' ? tooltip : '';
```

File: packages/grafana-ui/src/components/IconButton/IconButton.tsx [hunk 2]
```diff
@@ -49,9 +49,15 @@ export const IconButton = React.forwardRef<HTMLButtonElement, Props>(
...
 
     const button = (
-      <button ref={ref} aria-label={ariaLabel || tooltip || ''} {...restProps} className={cx(styles.button, className)}>
+      <button
+        ref={ref}
+        aria-label={ariaLabel || tooltipString || ''}
```
Comment (ashharrison90) [inline packages/grafana-ui/src/components/IconButton/IconButton.tsx:57]:
```suggestion
        aria-label={ariaLabel || tooltipString}
```

**Review by ashharrison90 — APPROVED** [review]
lgtm, minor type suggestions to avoid using `as` 😄

**Pull request merged**